src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java

```diff
@@ -0,0 +1,166 @@
+    private static final int BUCKET_FS_PORT = 2580;
+    private static final int DATABASE_PORT = 8563;
+    private final ConnectionDetailProvider connectionDetails;
+    private final String bucketFsReadPassword;
+    private final String bucketFsWritePassword;
+    private final SshConnection sshConnection;
+    private final int localBucketFsPort;
+    private final int localDatabasePort;
+    private final List<String> dataNodeIds;
+
+    /**
+     * Create a new instance of {@link StandaloneExasolTestSetup}.
+     * 
+     * @param connectionDetailProvider provider for the connection details
+     */
+    public StandaloneExasolTestSetup(final ConnectionDetailProvider connectionDetailProvider) {
+        this.connectionDetails = connectionDetailProvider;
+        this.sshConnection = setupSshConnection();
+        this.dataNodeIds = fetchDataNodeIds();
+        this.localBucketFsPort = this.sshConnection.addForwardPortForwarding(BUCKET_FS_PORT, getADataNode());
+        final int localHttpsPort = this.sshConnection.addForwardPortForwarding(443);
+        this.localDatabasePort = this.sshConnection.addForwardPortForwarding(DATABASE_PORT, getADataNode());
+        final ExaOperationInterface exaOperationInterface = new ExaOperationInterface("localhost:" + localHttpsPort,
+                connectionDetailProvider.getAdminCredentials());
+        exaOperationInterface.startStorageServiceIfNotRunning();
+        exaOperationInterface.startAllDatabases();
+        exaOperationInterface.setBucketFsPort("bfsdefault", BUCKET_FS_PORT);
+        this.bucketFsReadPassword = generatePassword();
+        this.bucketFsWritePassword = generatePassword();
+        exaOperationInterface.setBucketPasswords(this.bucketFsReadPassword, this.bucketFsWritePassword);
+        waitForExasolToSyncTheSettings();
+        waitUntil(this::isSqlInterfaceAvailable);
+        cleanTheDatabase();
+    }
+
+    private String getADataNode() {
+        if (this.dataNodeIds.isEmpty()) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-ETAJ-27").message("No datanodes available.")
+                    .mitigation("Probably the cluster has not started completely. Try to wait a bit.").toString());
+        }
+        return this.dataNodeIds.get(0);
+    }
+
+    private List<String> fetchDataNodeIds() {
+        final List<String> nodes = new ArrayList<>(
+                Arrays.asList(this.sshConnection.runCommand("/usr/opt/EXASuite-7/EXAClusterOS-7.0.8/bin/cosinfo -e")
+                        .whenFinished().assertExitCodeIsZero().getStdout().split("\n")));
+        nodes.removeIf(nodeId -> nodeId.equals("10"));// remove management node
+        return nodes.stream().map(id -> "n" + id).collect(Collectors.toList());
+    }
+
+    private SshConnection setupSshConnection() {
+        SshConnection connection = new SshConnection(this::configSshAuth);
+        connection = addGatewayPortsIfRequired(connection);
```

Comment:
```suggestion
        addGatewayPortsIfRequired(connection);
```

Reply:
we need to re-assign here since the method restarts the ssh server and the creates a new connection